### PR TITLE
feat(icon): add icon for NVIDIA GPU

### PR DESCRIPTION
**Pull request by radireddy**
show NVIDIA GPU icon for NVIDIA GPU Cloud

Closes #MBP-154

### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

## Title of the Pull Requests
Support NVIDIA GPU OS (NVIDIA GPU Cloud) while creating VM

### Description of the Change
Show NVIDIA GPU OS in  preinstalled applications and distribution + apps, only for T1 instances
Add below validations
1. NVIDIA OS can only be used with T1 instances
2. Can not update the existing server (without T1 instance) with NVIDIA OS image

### Benefits
The customer can preinstall NVIDIA GPU OS in VM

### Possible Drawbacks
none

### Applicable Issues
MBP-154

### CODE REVIEW

File: client/app/cloud/project/compute/infrastructure/virtualMachine/addEdit/cloud-project-compute-infrastructure-virtualMachine-addEdit.controller.js [hunk 1]
```diff
@@ -376,6 +376,26 @@ angular.module('managerApp')
             self.states.hasOldFlavors = true;
           }
         });
+        // Remove flavors if OS has restricted
+        const restrictedFlavors = _.get(self.vmInEdition, 'image.flavorType') || [];
```
Comment (marie-j) [inline client/app/cloud/project/compute/infrastructure/virtualMachine/addEdit/cloud-project-compute-infrastructure-virtualMachine-addEdit.controller.js]:
```suggestion
         const restrictedFlavors = _.get(self.vmInEdition, 'image.flavorType', []);
```

File: client/app/cloud/project/compute/infrastructure/virtualMachine/addEdit/cloud-project-compute-infrastructure-virtualMachine-addEdit.controller.js [hunk 2]
```diff
@@ -376,6 +376,26 @@ angular.module('managerApp')
...
+        if (restrictedFlavors.length > 0) {
+          self.displayData.categories = _.filter(
+            self.displayData.categories,
+            (category) => {
+              // eslint-disable-next-line no-param-reassign
```
Comment (marie-j) [inline client/app/cloud/project/compute/infrastructure/virtualMachine/addEdit/cloud-project-compute-infrastructure-virtualMachine-addEdit.controller.js]: Please avoid disabling eslint

File: client/app/cloud/project/compute/infrastructure/virtualMachine/addEdit/cloud-project-compute-infrastructure-virtualMachine-addEdit.controller.js [hunk 3]
```diff
@@ -376,6 +376,26 @@ angular.module('managerApp')
...
+              category.flavors = _.filter(
+                category.flavors,
+                flavor => _.indexOf(restrictedFlavors, flavor.shortType) > -1,
+              );
+              return category.flavors.length > 0;
+            },
+          );
```
Comment (marie-j) [inline client/app/cloud/project/compute/infrastructure/virtualMachine/addEdit/cloud-project-compute-infrastructure-virtualMachine-addEdit.controller.js]:
```suggestion
           self.displayData.categories = self.displayData.categories.filter(category => 
            category.flavors.some(flavor => restrictedFlavors.includes(flavor.shortType)))
```

Comment (radireddy) [reply client/app/cloud/project/compute/infrastructure/virtualMachine/addEdit/cloud-project-compute-infrastructure-virtualMachine-addEdit.controller.js]: this suggestion does not work for my use case

**Comment by radireddy** [comment]
waiting for MBP-247 to be resolved

File: client/app/cloud/project/compute/infrastructure/virtualMachine/addEdit/cloud-project-compute-infrastructure-virtualMachine-addEdit.controller.js [hunk 4]
```diff
@@ -1436,6 +1467,21 @@ angular.module('managerApp')
...
+       * If previous flavour is not t1 or t2, dont allow to select NVIDIA.
+       * This is also applicable for other images with restricted flavours.
+       */
+      self.isImageSelectableInEditMode = function isImageSelectableInEditMode(image) {
+        const flavorTypes = _.get(image, 'flavorType', []);
+        if (self.vmInEdition.status === 'DRAFT' || _.isEmpty(flavorTypes)) {
+          return true;
+        }
+        const flavorType = _.get(self.vmInEdition, 'flavor.shortType', null);
+        return _.indexOf(flavorTypes, flavorType) !== -1;
```
Comment (JeremyDec) [inline client/app/cloud/project/compute/infrastructure/virtualMachine/addEdit/cloud-project-compute-infrastructure-virtualMachine-addEdit.controller.js]: You can : 
```suggestion
        return _.includes(flavorTypes, flavorType);
```

File: client/app/cloud/project/compute/infrastructure/virtualMachine/addEdit/cloud-project-compute-infrastructure-virtualMachine-addEdit.controller.js [hunk 5]
```diff
@@ -376,6 +376,37 @@ angular.module('managerApp')
...
+        let selectedFlavourInFilterList = false;
+        if (!_.isEmpty(restrictedFlavors)) {
+          self.displayData.categories = _.filter(
+            self.displayData.categories,
+            (category) => {
+              const currentCategory = category;
+              currentCategory.flavors = _.filter(
+                category.flavors,
+                (flavor) => {
+                  const isRestricted = _.indexOf(restrictedFlavors, flavor.shortType) > -1;
```
Comment (JeremyDec) [inline client/app/cloud/project/compute/infrastructure/virtualMachine/addEdit/cloud-project-compute-infrastructure-virtualMachine-addEdit.controller.js]: You can : 
```suggestion
const isRestricted = _.includes(restrictedFlavors, flavor.shortType);
```

**Comment by jleveugle** [comment]
Hello @radireddy,
Your pull request is now closed due to inactivity.
Feel free to re-open it if you have any update.
Anna Bot for OVH UX <3

**Comment by radireddy** [comment]
still blocked by API

**Comment by FredericEspiau** [comment]
You should rebase on `develop`

**Comment by radireddy** [comment]
new PR created to merge these changes
https://github.com/ovh-ux/ovh-manager-cloud/pull/1277

**Pull request closed without merging**